apitools/base/py/exceptions.py

```diff
@@ -74,9 +74,48 @@ def status_code(self):
         return int(self.response['status'])
 
     @classmethod
-    def FromResponse(cls, http_response):
-        return cls(http_response.info, http_response.content,
-                   http_response.request_url)
+    def FromResponse(cls, http_response, method_config=None, request=None):
+        try:
+            status_code = int(http_response.info.get('status'))
+            error_cls = _HTTP_ERRORS.get(status_code, cls)
+        except ValueError:
+            error_cls = cls
+        return error_cls(http_response.info, http_response.content,
+                         http_response.request_url, method_config, request)
+
+
+class HttpBadRequestError(HttpError):
+
```

Comment:
I'd drop the newlines between class definition and docstring.

Reply:
Done! (I originally kept them for consistency with the rest of the file, but I changed it now)